Libraries/LibWeb/HTML/XMLSerializer.cpp

```diff
@@ -769,8 +769,20 @@ static WebIDL::ExceptionOr<String> serialize_comment(DOM::Comment const& comment
 // https://w3c.github.io/DOM-Parsing/#xml-serializing-a-text-node
 static WebIDL::ExceptionOr<String> serialize_text(DOM::Text const& text, [[maybe_unused]] RequireWellFormed require_well_formed)
 {
-    // FIXME: 1. If the require well-formed flag is set (its value is true), and node's data contains characters that are not matched by the XML Char production,
-    //           then throw an exception; the serialization of this node's data would not be well-formed.
+    // 1. If the require well-formed flag is set (its value is true), and node's data contains characters that are not matched by the XML Char production,
+    //    then throw an exception; the serialization of this node's data would not be well-formed.
+    if (require_well_formed == RequireWellFormed::Yes) {
+        auto const& data = text.data();
+        for (auto it = data.code_points().begin(); it != data.code_points().end(); ++it) {
```

Comment:
This is minor, but we can change this to a range-based for loop which is a bit less code / a bit nicer to read:
```cpp
        for (u32 code_point : text.data().code_points()) {
```

However, having noticed that, from looking around this file there are other places that we need to check against the XML Char production, so it would be nice that were moved into a helper function so this is easy to call in the other places for when that is also implemented